src/providers/ldap/ldap_id.c

```diff
@@ -292,7 +297,13 @@ struct tevent_req *users_get_send(TALLOC_CTX *memctx,
         }
     }
 
-    if (state->use_id_mapping || filter_type == BE_FILTER_SECID) {
+    if (state->non_posix == true) {
+        state->filter = talloc_asprintf(state,
```

Comment:
I think the only reason is that I work too much with Andreas :-) https://blog.cryptomilk.org/2013/03/28/writing-and-reading-code/

I'll change that to the short form.